src/session.rs

```diff
@@ -341,13 +346,9 @@ where
                 (session.is_destroyed(), session.data_changed());
             drop(session);
```

Comment:
Do you need this `drop(session)` anymore?

Reply:
Good catch!